pos_receipt_hide_user/models/pos_config.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class PosConfig(models.Model):
+    _inherit = "pos.config"
+
+    hide_user = fields.Boolean(string="Hide User in POS receipt?", default=True)
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    hide_user = fields.Boolean(related="pos_config_id.hide_user", readonly=False)
```

Comment:
Nitpicking / not-blocking

```suggestion
    pos_hide_user = fields.Boolean(related="pos_config_id.hide_user", readonly=False)
```

Since `res.config.settings` is a global model, used for all kinds of settings, having a prefix here would decrease the risk of overlapping with other modules